api/variables.go

```diff
@@ -124,87 +124,98 @@ func (sv *Variables) Delete(path string, qo *WriteOptions) (*WriteMeta, error) {
 	path = cleanPathString(path)
-	wm, err := sv.deleteChecked(path, checkIndex, qo)
+	wm, err := vars.deleteChecked(path, checkIndex, qo)
 	if err != nil {
 		return nil, err
 	}
-
 	return wm, nil
 }
 
 // List is used to dump all of the variables, can be used to pass prefix
 // via QueryOptions rather than as a parameter
-func (sv *Variables) List(qo *QueryOptions) ([]*VariableMetadata, *QueryMeta, error) {
-
+func (vars *Variables) List(qo *QueryOptions) ([]*VariableMetadata, *QueryMeta, error) {
 	var resp []*VariableMetadata
-	qm, err := sv.client.query("/v1/vars", &resp, qo)
+	qm, err := vars.client.query("/v1/vars", &resp, qo)
 	if err != nil {
 		return nil, nil, err
 	}
 	return resp, qm, nil
 }
 
 // PrefixList is used to do a prefix List search over variables.
-func (sv *Variables) PrefixList(prefix string, qo *QueryOptions) ([]*VariableMetadata, *QueryMeta, error) {
-
+func (vars *Variables) PrefixList(prefix string, qo *QueryOptions) ([]*VariableMetadata, *QueryMeta, error) {
 	if qo == nil {
 		qo = &QueryOptions{Prefix: prefix}
 	} else {
 		qo.Prefix = prefix
 	}
-
-	return sv.List(qo)
+	return vars.List(qo)
 }
 
-// GetItems returns the inner Items collection from a variable at a
-// given path
-func (sv *Variables) GetItems(path string, qo *QueryOptions) (*VariableItems, *QueryMeta, error) {
+// GetItems returns the inner Items collection from a variable at a given path.
+//
+// Deprecated: Use GetVariableItems instead.
+func (vars *Variables) GetItems(path string, qo *QueryOptions) (*VariableItems, *QueryMeta, error) {
+	vi, qm, err := vars.GetVariableItems(path, qo)
+	if err != nil {
+		return nil, nil, err
+	}
+	return &vi, qm, nil
+}
 
+// GetVariableItems returns the inner Items collection from a variable at a given path.
+func (vars *Variables) GetVariableItems(path string, qo *QueryOptions) (VariableItems, *QueryMeta, error) {
 	path = cleanPathString(path)
-	svar := new(Variable)
+	v := new(Variable)
 
-	qm, err := sv.readInternal("/v1/var/"+path, &svar, qo)
+	qm, err := vars.readInternal("/v1/var/"+path, &v, qo)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	return &svar.Items, qm, nil
+	// note: readInternal will in fact turn our v into a nil if not found
+	if v == nil {
+		return nil, nil, ErrVariablePathNotFound
+	}
+
+	return v.Items, qm, nil
 }
 
 // readInternal exists because the API's higher-level read method requires
 // the status code to be 200 (OK). For Peek(), we do not consider 403 (Permission
 // Denied or 404 (Not Found) an error, this function just returns a nil in those
 // cases.
-func (sv *Variables) readInternal(endpoint string, out **Variable, q *QueryOptions) (*QueryMeta, error) {
+func (vars *Variables) readInternal(endpoint string, out **Variable, q *QueryOptions) (*QueryMeta, error) {
+	// todo(shoenig): seems like this could just return a *Variable instead of taking
+	// in a **Variable and modifying it?
```

Comment:
I went back to look at the original PR for this and don't see any discussion of it. Maybe @angrycub will remember why we did this?